pkg/preparer/orchestrate.go

```diff
@@ -79,6 +81,90 @@ func (p podWorkerID) String() string {
 	return fmt.Sprintf("%s-%s", p.podID.String(), p.podUniqueKey)
 }
 
+func (p *Preparer) InstallRequiredPods(requiredPods map[types.PodID]bool, preparerConfig *PreparerConfig) error {
+	intentResults, _, err := p.store.ListPods(consul.INTENT_TREE, p.node)
+	var intentRequired []consul.ManifestResult
+	if err != nil {
+		p.Logger.WithError(err).Errorln("could not check intent")
+	}
+	errorChan := make(chan error)
+	quit := make(chan struct{})
+
+	for _, intentResult := range intentResults {
+		if _, ok := requiredPods[intentResult.Manifest.ID()]; ok {
+			// check pod health here, to prevent reinstall. This is useful when
+			// the required pod is already running, and the p2-prepare restarted
+			if p.checkPodHealth(intentResult, preparerConfig) {
```

Comment:
why does the normal intent/reality diff not catch this?

Reply:
This happens before p2-prepare normally start actually. So p2-prepare not watching the intent store now. I planed to make p2-prepare to finish install all required pods, then start the thread to watch for intent/reality diffs.